app/views/admin/versions/_object_desc_and_link.html.haml

```diff
@@ -1,156 +1,171 @@
-                  id: commercialable_last_version.id, anchor: 'commercials-content')
+    commercial in venue
+    - if commercialable_last_version.item
+      = link_to commercialable.name,
+            edit_admin_conference_venue_path(conference_id: Conference.find(version.conference_id).short_title,
+                                                id: commercialable.id, anchor: 'commercials-content')
     - else
-      = 'commercial'
-    = "in venue #{commercialable_last_version.name}"
+      = commercialable.name
 
   - when 'Conference'
     = link_to 'commercial',
             admin_conference_commercials_path(conference_id: Conference.find(version.conference_id).short_title)
 
-- elsif %w(EventsRegistration Comment Vote).include?(version.item_type)
+- when 'EventsRegistration', 'Comment', 'Vote', 'Event'
   = 'event'
-  - event_id = get_version_object(version).try(:event_id) || get_version_object(version).try(:commentable_id)
-  = link_to Event.find(event_id).title,
+  - object = current_or_last_object_state(version.item_type, version.item_id)
+  - event_id = object.try(:id) || object.try(:event_id) || object.commentable_id
+  = link_to (current_or_last_object_state('Event', event_id).try(:title) || 'deleted event'),
           admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title, id: event_id)
 
-- elsif version.item_type =='Target'
+- when 'Target'
   = 'target'
-  - if version.item
-    = link_to Target.find(version.item_id).to_s,
-            admin_conference_targets_path(conference_id: Conference.find(version.conference_id).short_title)
-  - else
-    = PaperTrail::Version.where(item_type: 'Target', item_id: version.item_id).last.reify.to_s
-
-- elsif  version.item
-  - case version.item_type
-  - when 'Conference'
-    = 'conference'
-    = link_to Conference.find(version.conference_id).title,
-            edit_admin_conference_path(id: Conference.find(version.conference_id).short_title)
-
-  - when 'RegistrationPeriod'
-    = link_to 'registration period',
-            admin_conference_registration_period_path(conference_id: Conference.find(version.conference_id).short_title)
-
-  - when 'Contact'
-    = link_to 'contact details',
-            edit_admin_conference_contact_path(conference_id: Conference.find(version.conference_id).short_title)
+  - target = current_or_last_object_state(version.item_type, version.item_id)
+  = link_if_alive version, target.to_s, admin_conference_targets_path(conference_id: Conference.find(version.conference_id).short_title)
+
+- when 'EventSchedule'
+  - event_schedule = current_or_last_object_state(version.item_type, version.item_id)
+  event
+  = link_to (current_or_last_object_state('Event', event_schedule.event_id).try(:title) || 'deleted event'),
+           admin_conference_program_event_path(conference_id: Conference.find(version.conference_id).short_title, id: event_schedule.event_id)
```

Comment:
We still seem to have some wrong links being created here for Event but I really want to keep going with the rest of the PRs, so I am putting this on a note here to fix in a following PR.